packages/e2e/src/specs/messages/messages-thread-creation.e2e.ts

```diff
@@ -0,0 +1,331 @@
+import { waitForClickableAndClick } from "../../helpers/general.helper"
+import {
+  deleteConversationOnThreadList,
+  sendMessage,
+} from "../../helpers/messages.helper"
+import messagesConversationPage from "../../page-objects/messages-conversation.page"
+
+describe("New thread creation scenarios - no contacts & no threads", () => {
+  before(async () => {
+    // Waiting for device connected through USB
+    browser.pause(10000)
+
+    await ModalGeneralPage.clickUpdateAvailableModalCloseButton()
+
+    await waitForClickableAndClick(await NavigationTabs.messagesTab)
+  })
+  it("Press browse contacts button and check no contacts is displayed", async () => {
+    await waitForClickableAndClick(await MessagesPage.newMessageButton)
+    await waitForClickableAndClick(
+      await MessagesConversationPage.browseContactsButton
+    )
+
+    const searchInput = await BrowseContactsModal.searchInput
+    await searchInput.waitForDisplayed()
+
+    const contactList = await BrowseContactsModal.emptyContactList
+    await expect(contactList).toBeDisplayed()
+  })
+
+  it("Check modal title is Browse Contacts and close it", async () => {
+    const modalHeader = await BrowseContactsModal.modalHeader
+    await expect(modalHeader).toHaveText("Browse Contacts")
+
+    await BrowseContactsModal.closeModalButtonClick()
+  })
+
+  it("Input message text first then recipient number and check message Send button is clickable", async () => {
+    const testMessage = "First input message text"
+    const testNumber = "11111"
+    await MessagesConversationPage.insertTextToMessageInput(testMessage)
+    await MessagesConversationPage.insertTextToSearchContactInput(testNumber)
+
+    const sendButton = await MessagesConversationPage.sendMessageButton
+    await expect(sendButton).toBeClickable()
+  })
+
+  it("Delete existing recipient number and insert new one", async () => {
+    const testNumber = "22222"
+    await waitForClickableAndClick(
+      await MessagesConversationPage.messageTextInput
+    )
+    await waitForClickableAndClick(
+      await MessagesConversationPage.searchContactsInput
+    )
+    const recipientInputValue =
+      await MessagesConversationPage.searchContactsInput.getValue()
+    for (let i = 0; i <= (await recipientInputValue).length; i++) {
+      await browser.keys("\ue003")
+    }
+    await MessagesConversationPage.insertTextToSearchContactInput(testNumber)
+
+    const recipientInput = await MessagesConversationPage.searchContactsInput
+    await expect(recipientInput).toHaveValue(testNumber)
+  })
+  it("Check thread name is 'New conversation'", async () => {
+    const thaedName = await MessagesPage.threadRow
+    await expect(thaedName).toHaveText("New Conversation")
+  })
+
+  it("Click on recipient input field and send Enter key to lock it and create a thread", async () => {
+    waitForClickableAndClick(await MessagesConversationPage.messageTextInput)
+    await browser.keys("\uE007")
```

Comment:
key to constant :)

Reply:
https://github.com/mudita/mudita-center/pull/1160/commits/ddde3cbe49af11bdbf4cfc913bce7fdfa19f39a6